packages/fhir-location-management/src/helpers/utils.ts

```diff
@@ -14,16 +14,19 @@ import { HTTPError } from '@opensrp/server-service';
  * Parse the raw child hierarchy node map
  *
  * @param rawNodeMap - Object of raw hierarchy nodes
+ * @param admLevel - location administrative level
  * @returns Array of Parsed hierarchy nodes
  */
-const parseFhirChildren = (rawNodeMap: ChildNodeList[]) => {
+const parseFhirChildren = (rawNodeMap: ChildNodeList[], admLevel = 0) => {
   return rawNodeMap.map((child) => {
     // standardize the parsedHierarchy node structure to be similar for both
     // root node and all other descendant nodes.
     const { treeNode } = child;
+    const nextAdmLevel = admLevel + 1;
     const parsedNode: ParsedHierarchyNode = {
       ...treeNode,
-      children: parseFhirChildren(treeNode.children ?? []) as ParsedHierarchyNode[],
+      administrativeLevel: admLevel,
+      children: parseFhirChildren(treeNode.children ?? [], nextAdmLevel) as ParsedHierarchyNode[],
```

Comment:
I feel iffy about adding the adminstrative level here for 2 reasons:
1. it possibly duplicates information that already exists in the node
2. There is no guarantee that `parseFhirChildren` will only always be used to generate a tree starting with the the root fhir location.